conda/environments/builddocs.yml

```diff
@@ -16,5 +16,15 @@ dependencies:
 - pip
 - pip:
   - --extra-index-url=https://pypi.anaconda.org/rapidsai-wheels-nightly/simple
+  # NOTE: this 'pip install' working correctly depends on two environment variables
+  #       used by rapids-build-backend, set in the readthedocs UI
+  #
+  #       # do not try to run 'nvcc --version' to determine CUDA version
+  #       RAPIDS_DISABLE_CUDA=true
+  #
+  #       # install CUDA-12-specific dependencies
+  #       # (e.g. libucx-cu12)
+  #       RAPIDS_MATRIX_ENTRY="cuda=12.2"
+  #
   - ../../
```

Comment:
It looks like we use a conda env with a pip dependency on `../..`. I think it would be clearer to install a conda env with the relevant dependencies and then invoke `pip install` with the appropriate flags on the command line instead of requiring environment variables in the UI. Is that an option?

Reply:
It's not an option. `readthedocs` doesn't allow you to customize the flags passed to `pip install`.

You can see the history that led to this pattern here: https://github.com/rapidsai/ucx-py/pull/1041#discussion_r1592789721

Reply:
Hmm. I saw there were options to run arbitrary commands like installing with `poetry` or `uv`, so I figured we could do the same with `pip` and custom flags. https://docs.readthedocs.io/en/stable/build-customization.html#install-dependencies-with-poetry

Reply:
!!! in all of that conversation in #1041, I was so focused on "how do we change the `install:` step", I never considered the possibility "just ignore it completely and install the library through one of the other run custom commands" options. The tunnel vision was real 😭 

Thanks for that link, let me try this.

Reply:
This worked! I can't believe I missed that going through options in #1041, thank you so much 😅 

I removed the environment variables in the readthedocs console, then pushed the changes you now see in `.readthedocs.yaml` on this PR.

* build: https://readthedocs.org/projects/ucx-py/builds/24674174/
* docs: https://ucx-py--1044.org.readthedocs.build/en/1044/

I think this is unambiguously better than having the `pip:` section in a conda environment YAML file.

Reply:
![Aw Yeah](https://i.giphy.com/media/v1.Y2lkPTc5MGI3NjExYmhvMGpndGgyMGhpajI3bjd0cnhwMGFmMzVjbjZ4dHo0MDl4eWR3MyZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/3o84UaGEtyayvBIFwc/giphy.gif)